Doc/library/urllib.request.rst

```diff
@@ -1323,15 +1328,18 @@ Use the *headers* argument to the :class:`Request` constructor, or::
    req.add_header('Referer', 'http://www.python.org/')
    # Customize the default User-Agent header value:
    req.add_header('User-Agent', 'urllib-example/0.1 (Contact: . . .)')
-   r = urllib.request.urlopen(req)
+   with urllib.request.urlopen(req) as r:
+       print(r.read().decode('utf-8'))
+
 
 :class:`OpenerDirector` automatically adds a :mailheader:`User-Agent` header to
 every :class:`Request`.  To change this::
 
    import urllib.request
    opener = urllib.request.build_opener()
    opener.addheaders = [('User-agent', 'Mozilla/5.0')]
-   opener.open('http://www.example.com/')
+   with opener.open('http://www.example.com/') as response:
+      print(response.read().decode('utf-8'))
```

Comment:
```suggestion
   with opener.open('http://www.example.com/') as r:
      print(r.read().decode('utf-8'))
```

I would either use 'res' or just 'r' as in the previous example. Maybe 'r' is better but 'res' shows that it's a "response".